test/ext.lean

```diff
@@ -76,6 +76,68 @@ example (f : ℕ × (ℕ → ℕ)) : f = f := by
 example (f : Empty → Empty) : f = f := by
   ext ⟨⟩
 
+
+
+/-!
+## Test for `ext?`
+It should apply the two ext lemmas and display a tactic replacement
+-/
+namespace nested
+
+structure A where
+  a : Nat
+structure B extends A where
+  b : Nat
+structure C extends B where
+  c : Nat
+
+@[ext]
+theorem extCtoB (c₁ c₂ : C) (g : ∀ (a : A), true)  (h : c₁.c = c₂.c) (h' :  c₁.toB = c₂.toB ) :
+    c₁ = c₂ := by admit
+@[ext]
+theorem extBtoA (b₁ b₂ : B) (h : b₁.b = b₂.b) (h' :  b₁.a = b₂.a ) : b₁ = b₂ := by admit
+
+@[ext]
+theorem test (c₁ c₂ : C) (h : c₁.a = c₂.a) (h' :  c₁.b = c₂.b )
+    (h'' :  c₁.c = c₂.c ) : c₁ = c₂ := by
+  ext?
+  -- Try this:
+  -- · apply extCtoB
+  --   intros
+  --   sorry
+  --   sorry
+  --   apply extBtoA
+  --   sorry
+  --   sorry
```

Comment:
Can we teach this to use `·` to group subgoals?

Reply:
With the current design of using a `IO.mkRef #[]` to keep track of the used lemmas, I believe this would be very hard. I tried once and figured it wasn't worth the effort back then.

Reply:
Can we just use `show_term`, and print a `refine` statement with `?_` holes? Seems easier, and gives better structured output.

Reply:
Interesting! Maybe we should decipher that first. Could you post an example on zulip and ping me?